### PR TITLE
tools:add VimCalc plugin

### PR DESCRIPTION
**Pull request by tracyone**
SPC + ac open calculator

change:
SPC + al open calendar

# PR Prelude

Thank you for working on SpaceVim! :)

**Please complete these steps and check these boxes (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [ x] I have read and understood SpaceVim's [CONTRIBUTING][cont] document.
- [ x] I have read and understood SpaceVim's [CODE_OF_CONDUCT][code] document.
- [ ] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [ x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

VimCalc is the most powerful calculator plugin for vim that I ever seen

[cont]: https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md
[code]: https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md

### CODE REVIEW

**Comment by wsdjeg** [comment]
for apps, I think we should not change the key mappings, and the last key binding conventions:

1. do not use same keys if possible, this is to avoid double click, for example we can not use `SPC a a`
2. old key binding can not be change if it works well.


for this pr, as we has already added `SPC a c` for **c**alendar, then if we need to add new app `calculator`, so we need to use `SPC a l` for it as `SPC a a` is avoid.


and no need to add two plugins at same time, it will make dein slow.

you can use:

```vim
let s:CMP = SpaceVim#api#import('vim#compatible')
if s:CMP.has('python')
  call add(plugins, ['gregsexton/VimCalc', {'on_cmd' : 'Calc'}])
elseif s:CMP.has('python3')
  call add(plugins, ['fedorenchik/VimCalc3', {'on_cmd' : 'Calc'}]
endif
```

in vim, the `has('python')` is not enough for checking python support.
if

**Review by wsdjeg — COMMENTED** [review]
Thanks for your pr.

**Comment by tracyone** [comment]
日历功能对比计算器功能太少用了，ac这个后缀，我觉得应该给计算器用，这是我的理由。

我一般不推荐，推荐必属精品，这货实际体验不会输给系统自带的计算器。

**Comment by wsdjeg** [comment]
ok, 但是这个代码需要改改呢，同时将两个插件都加入列表，虽然有 `if` option 限制，但是还是增加dein的负担。 当 vim 不支持 py/py3 时应当不要定义这个快捷键。并且，单独用 vim 自代的 has 来检测py支持很不准确，经常会处问题，有时候 has python 返回1,但是计算 ：py 命令并不能用。

**Comment by tracyone** [comment]
done

File: autoload/SpaceVim/layers/tools.vim [hunk 1]
```diff
@@ -59,6 +69,8 @@ function! SpaceVim#layers#tools#config() abort
   augroup rainbow_lisp
     autocmd!
     autocmd FileType lisp,clojure,scheme,java RainbowParentheses
+    autocmd FileType vimcalc setlocal nonu nornu nofoldenable | imap <buffer> <c-d> :q<cr>
+          \ | nmap <buffer> q :bdelete<cr>
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/tools.vim]: here you need to use `nnoremap` and `inoremap`, if someone map any char in `:bdelete<cr>`, this mapping will not work.

**Comment by tracyone** [comment]
```vim
inoremap <buffer> <c-d> :q<cr>
```
这个不起作用
但是这个可以：

```vim
imap <buffer> <c-d> :q<cr>
```
不知道为什么

**Comment by wsdjeg** [comment]
@tracyone 因为 你加的这个插件 定义了 一个 `inoremap :q<cr> <esc>:q<cr>`, 或者类似的。

而你定义的 `imap <c-d>` 递归解释了 以上的 隐射。

你可以直接用：

```vim
inoremap <buffer> <c-d> <esc>:q<cr>
```

**Comment by tracyone** [comment]
ok, I will fix it tonight.

**Comment by tracyone** [comment]
done

**Review by wsdjeg — APPROVED** [review]

**Comment by wsdjeg** [comment]
@tracyone I just approved this PR, and I have invite you to join us, if you join our developers team, you will be able to merge approved PR yourself. :)

**Pull request merged**